### PR TITLE
Fix macos from source

### LINKED ISSUE #127: sklearn>=0.22 and Py3.8 compat: sklearn.preprocessing.Imputer has been removed
scikit-learn deprecated the `sklearn.preprocessing.Imputer` in `0.20.0` and removed it in `0.22.0`:
https://github.com/scikit-learn/scikit-learn/blame/0.22.1/doc/whats_new/v0.20.rst#L1444-L1468

This means `py_entitymatching` is incompatible with `scikit-learn >=0.22` due to
https://github.com/anhaidgroup/py_entitymatching/blob/v0.3.2/py_entitymatching/matcher/matcherutils.py#L11
and
https://github.com/anhaidgroup/py_entitymatching/blob/v0.3.2/py_entitymatching/matcher/matcherutils.py#L221-L224

This in turn means `py_entitymatching` does not fully work with Python 3.8 since `scikit-learn <0.22` does not support the latter.
ref: https://github.com/conda-forge/py_entitymatching-feedstock/pull/3

The above referenced `whats_new/v0.20.rst` mentions `sklearn.impute.SimpleImputer`, `sklearn.preprocessing.FunctionTransformer` (for the `axis` keyword), and `numpy.nan` (for the `missing_values` keyword) as replacements.
I have no experience with scikit-learn so am not able to offer further assistance/PRs, unfortunately.

### PR DESCRIPTION
**Pull request by christiemj09**
The build on macOS from source has been broken for a while now. It's because recent versions of conda have dropped gcc as an available package (there are other related packages):

```bash
# Known version of conda with gcc available
docker container run --name conda4.4 continuumio/miniconda:4.4.10 conda search gcc

# Latest version of conda
docker container run --name condalatest continuumio/miniconda:latest conda search gcc
```

This PR simply drops the conditional installation of gcc on macOS; the latest macOS instances on Travis have gcc installed.

NOTE: As a temporary solution, this PR also contains pinned numpy and scikit-learn versions prior to installing dependencies. This is currently necessary in order to support Python 2.7 and to avoid #127. A long-term solution should fix #127 and drop support for Python 2.

### CODE REVIEW

**Comment by jatinarora2409** [comment]
This is a temporary fix, where we hardcode the old versions of NumPy and sckit-learn. We should work on making the entity matching module.
